FBSnapshotTestCase/FBSnapshotTestController.m

```diff
@@ -226,6 +228,15 @@ - (NSString *)_fileNameForSelector:(SEL)selector
   if (0 < identifier.length) {
     fileName = [fileName stringByAppendingFormat:@"_%@", identifier];
   }
+  
+  if (self.isDeviceAgnostic) {
+    UIDevice *device = [UIDevice currentDevice];
+    NSString *model  = [device.model stringByReplacingOccurrencesOfString:@" " withString:@"_"];
+    NSString *os     = [device.systemVersion stringByReplacingOccurrencesOfString:@"." withString:@"_"];
```

Comment:
Have you considered using a broader replacement strategy here? Instead of searching for individual characters, you could do something like this:

```objc
NSMutableCharacterSet *invalidCharacters = [NSMutableCharacterSet new];
[invalidCharacters formUnionWithCharacterSet:[NSCharacterSet whitespaceCharacterSet]];
[invalidCharacters formUnionWithCharacterSet:[NSCharacterSet punctuationCharacterSet]];
// ...repeat with any other character sets you'd like to replace.
NSArray *validComponents = [self componentsSeparatedByCharactersInSet:invalidCharacters];
NSString *fileName = [validComponents componentsJoinedByString:@"_"];
```